src/mfbt/mozilla/CheckedInt.h

```diff
@@ -0,0 +1,800 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
```

Comment:
Please add comments to this file and IntegerTypeTraits, mentioning they are copy pasted from m-c and linking to the copied files.